lib/streamlit/elements/doc_string.py

```diff
@@ -31,26 +40,23 @@
 LOGGER: Final = get_logger(__name__)
 
 
-CONFUSING_STREAMLIT_MODULES: Final = (
-    "streamlit.echo",
-    "streamlit.delta_generator",
-    "streamlit.runtime.legacy_caching.caching",
-)
-
 CONFUSING_STREAMLIT_SIG_PREFIXES: Final = ("(element, ",)
 
 
 class HelpMixin:
     @gather_metrics("help")
-    def help(self, obj: Any) -> "DeltaGenerator":
-        """Display object's doc string, nicely formatted.
+    def help(self, obj: Any = None) -> "DeltaGenerator":
+        """Display help and other information for a given object.
 
-        Displays the doc string for this object.
+        Depending on the type of object that is passed in, this displays the
+        object's name, type, value, signature, docstring, and member variables,
+        methods -- as well as the values/docstring of members and methods.
 
         Parameters
         ----------
-        obj : Object
-            The object whose docstring should be displayed.
+        obj : any
+            The object whose information should be displayed. If `None`, this
+            call will display help for Streamlit itself.
 
         Example
```

Comment:
@snehankekre I'd like to embed apps below each of these examples, to show the output of `st.help()` in each case. How do we do that nowadays?

Reply:
@sfc-gh-tteixeira The process is described in [How to embed apps in our docs](https://www.notion.so/How-to-embed-apps-in-our-docs-ea6b286e759a45caaf5769a3d0184ad2). According to the process, if the behavior of the app to be embedded depends on some features in this PR branch (i.e. not in `streamlit-nightly`), we should first merge this feature PR. After merging, we wait for the feature to be available in nightly and then follow the process described in that doc. 

The embedded apps will have to come as a follow up PR. I can take of adding them and will mark you as a reviewer on that PR.